typescript/src/deposit.ts

```diff
@@ -340,11 +407,13 @@ export async function calculateDepositScriptHash(
   witness: boolean
 ): Promise<Buffer> {
   const script = await assembleDepositScript(deposit)
-  // Parse the script from HEX string.
-  const parsedScript = bcoin.Script.fromRaw(Buffer.from(script, "hex"))
   // If witness script hash should be produced, SHA256 should be used.
   // Legacy script hash needs HASH160.
-  return witness ? parsedScript.sha256() : parsedScript.hash160()
+  if (witness) {
+    return computeSha256(Hex.from(script)).toBuffer()
+  }
+
+  return Buffer.from(computeHash160(script), "hex")
```

Comment:
Just a nit: Can we keep it as ternary operator or at least a full `if & else`? Both options are equal paths but in the current form, the `witness` path looks like something exceptional. Not a big deal but it makes the code a bit more reader-friendly.

Reply:
Done in 332b0e772cc2eb9f5a61b678cb51a4ad15d16774.